### PR TITLE
 Fix having clause bug for complex joins

### LINKED ISSUE #2558: Aggregate in HAVING might use the wrong column in joins
```
CREATE TABLE descriptions (key int, description text);
CREATE TABLE amounts (key int, amount int);
SELECT create_distributed_table('descriptions','key');
SELECT create_distributed_table('amounts','key');

SELECT amounts.key, sum(amounts.amount)
FROM descriptions d1 JOIN descriptions d2 USING (key) JOIN amounts USING (key) 
GROUP BY amounts.key HAVING sum(amounts.amount) > 200;
...
WARNING:  function sum(text) does not exist
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
ERROR:  failed to execute task 16
```

Under the covers it's sending `sum(d1.description)` instead of `sum(amounts.amount)`:
```
COPY (SELECT amounts.key, sum(amounts.amount) AS sum, sum(amounts.amount) AS worker_column_3 FROM ((descriptions_103537 d1 JOIN descriptions_103537 d2 ON ((d1.key OPERATOR(pg_catalog.=) d2.key))) JOIN amounts_103569 amounts ON ((d1.key OPERATOR(pg_catalog.=) amounts.key))) WHERE true GROUP BY amounts.key HAVING (sum(d1.description) OPERATOR(pg_catalog.>) 200)) TO STDOUT
```

### PR DESCRIPTION
**Pull request by mtuncer**
We update column attributes of various clauses for a query
including target columns, select clauses when we introduce
new range table entries in the query.

It seems having clause column attributes were not updated.

This fix resolves the issue

DESCRIPTION: Fixes an issue with having clause when used with complex joins

Fixes #2558

### CODE REVIEW

**Comment by mtuncer** [comment]
@metdos sending your way for reviewer assignment

File: src/backend/distributed/planner/multi_physical_planner.c [hunk 1]
```diff
@@ -688,6 +688,18 @@ BuildJobQuery(MultiNode *multiNode, List *dependedJobList)
...
 
+	/* set correct column attributes for having columns */
+	if (updateColumnAttributes)
+	{
+		columnCell = NULL;
+		columnList = pull_var_clause_default((Node *) havingQual);
+		foreach(columnCell, columnList)
+		{
+			Var *column = (Var *) lfirst(columnCell);
+			UpdateColumnAttributes(column, rangeTableList, dependedJobList);
```
Comment (marcocitus) [inline src/backend/distributed/planner/multi_physical_planner.c]: This code looks almost identical to the code above.

Maybe:
```c
if (updateColumnAttributes)
{
    UpdateAllColumnAttributes(selectClauseList, rangeTableList, dependedJobList);
    UpdateAllColumnAttributes(havingQual, rangeTableList, dependedJobList);
}
```

Comment (mtuncer) [reply src/backend/distributed/planner/multi_physical_planner.c]: code is also duplicated earlier in the function for target entries

Comment (mtuncer) [reply src/backend/distributed/planner/multi_physical_planner.c]: will update all

**Pull request merged**